src/manufacturer/manufacturerDialog.component.tsx

```diff
@@ -160,19 +205,158 @@ function AddManufacturerDialog(props: AddManufacturerDialogProps) {
           setNameErrorMessage(
             'A manufacturer with the same name already exists.'
           );
+          return;
         }
+        setCatchAllError(true);
       });
-  }, [manufacturer, addManufacturer, handleClose]);
+  }, [handleErrors, manufacturer, addManufacturer, handleClose]);
+
+  const handleEditManufacturer = React.useCallback(() => {
+    if (selectedManufacturer && selectedManufacturerData) {
+      const hasErrors = handleErrors();
+
+      if (hasErrors) {
+        return;
+      }
+
+      const isNameUpdated = manufacturer.name !== selectedManufacturer.name;
+      const isURLUpdated =
+        manufacturer.url !== selectedManufacturer.url &&
+        manufacturer.url !== undefined;
+      const isBuildingNumberUpdated =
+        manufacturer.address?.building_number !==
+        selectedManufacturer.address.building_number;
+      const isStreetNameUpdated =
+        manufacturer.address?.street_name !==
+        selectedManufacturer.address.street_name;
+      const isTownUpdated =
+        manufacturer.address?.town !== selectedManufacturer.address.town;
+      const isCountyUpdated =
+        manufacturer.address?.county !== selectedManufacturer.address.county;
+      const ispostcodeUpdated =
+        manufacturer.address?.postcode !==
+        selectedManufacturer.address.postcode;
+      const isTelephoneUpdated =
+        manufacturer.telephone !== selectedManufacturer.telephone;
+
+      let ManufacturerToEdit: EditManufacturer = {
+        id: selectedManufacturer?.id,
+      };
+
+      if (isNameUpdated) {
+        ManufacturerToEdit = {
+          ...ManufacturerToEdit,
+          name: manufacturer.name,
+        };
+      }
+      if (isURLUpdated) {
+        ManufacturerToEdit = {
+          ...ManufacturerToEdit,
```

Comment:
Ah missed that this wasn't covered either, it seems there should be a unit test that checks the url can be updated.

Reply:
There is a test, just I think the url given to it is "Test", so its invalid which means this code doesn't run, so I think it just needs to be changed to a valid one. (https://github.com/ral-facilities/inventory-management-system/pull/112/files#diff-afc4e4a1b721a32857daaa89d191c8b99a161a5acefc3737abbdc4b8cdf69e41R583)